### PR TITLE
Improve PR merge check for CI

### PR DESCRIPTION
**Pull request by infinisil**
The Nixpkgs-vet workflow has been using a script to check whether the PR to check is mergeable and abort the check if it's not. This PR improves the script by:
- Making it reusable for other workflows. It can be used for any workflow checking out the PR really.
- Handle PRs that are not open anymore, fixing [this kind of misleading error](https://github.com/NixOS/nixpkgs/actions/runs/11279197077/job/31369348101#step:2:59):
  ```
  Checking whether the pull request can be merged
  GitHub is still computing whether this PR can be merged, waiting 5 seconds before trying again (4 retries left)
  [...]
  GitHub is still computing whether this PR can be merged, waiting 80 seconds before trying again (0 retries left)
  Checking whether the pull request can be merged
  Not retrying anymore. It's likely that GitHub is having internal issues: check https://www.githubstatus.com/
  ```
- Write some documentation for this script and how it can be used in workflows

### Things done

- [x] Tested that it works: https://github.com/tweag/nixpkgs/actions/runs/11280488261/job/31373578194?pr=97
- [x] Tested that it detects closed PRs: https://github.com/tweag/nixpkgs/actions/runs/11280512693/job/31373655841

---

This work is sponsored by [Antithesis](https://antithesis.com/) :sparkles:

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: ci/get-merge-commit.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# See ./README.md for docs
+
+set -euo pipefail
+
+log() {
+    echo "$@" >&2
+}
+
+repo=$1
+prNumber=$2
```
Comment (philiptaron) [inline ci/get-merge-commit.sh:11]: Want to do a check to see if that number of arguments are passed to the script?

File: ci/get-merge-commit.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    # Non-open PRs won't have their mergeability computed no matter what
+    state=$(jq -r .state <<< "$prInfo")
+    if [[ "$state" != open ]]; then
+        log "PR is not open anymore"
+        exit 1
+    fi
+
+    mergeable=$(jq -r .mergeable <<< "$prInfo")
+    mergedSha=$(jq -r .merge_commit_sha <<< "$prInfo")
```
Comment (philiptaron) [inline ci/get-merge-commit.sh:33]: Want to do this after testing for mergable?

**Review by philiptaron — APPROVED** [review]
Approved with a couple comments.

**Comment by infinisil** [comment]
@philiptaron Thanks, addressed!

**Review by philiptaron — APPROVED** [review]

**Pull request merged**